qiskit_ibm_runtime/base_runtime_job.py

```diff
@@ -91,49 +88,39 @@ def __init__(
         self._backend = backend
         self._job_id = job_id
         self._api_client = api_client
-        self._interim_results: Optional[Any] = None
         self._creation_date = creation_date
         self._program_id = program_id
         self._reason: Optional[str] = None
         self._reason_code: Optional[int] = None
         self._error_message: Optional[str] = None
         self._image = image
-        self._final_interim_results = False
         self._service = service
         self._session_id = session_id
         self._tags = tags
         self._usage_estimation: Dict[str, Any] = {}
         self._version = version
         self._queue_info: QueueInfo = None
-        self._user_callback = user_callback
         self._status: Union[RuntimeJobStatus, str] = None
 
         decoder = result_decoder or DEFAULT_DECODERS.get(program_id, None) or ResultDecoder
         if isinstance(decoder, Sequence):
-            self._interim_result_decoder, self._final_result_decoder = decoder
+            _, self._final_result_decoder = decoder
         else:
-            self._interim_result_decoder = self._final_result_decoder = decoder
-
-        # Used for streaming
-        self._ws_client_future = None  # type: Optional[futures.Future]
-        self._result_queue = queue.Queue()  # type: queue.Queue
-        self._ws_client = RuntimeWebsocketClient(
-            websocket_url=client_params.get_runtime_api_base_url().replace("https", "wss"),
-            client_params=client_params,
-            job_id=job_id,
-            message_queue=self._result_queue,
-        )
+            self._final_result_decoder = decoder
+
+        if user_callback or client_params:
+            warnings.warn(
+                "Interim results streaming was deprecated and removed in previous releases "
+                "so passing in 'user_callback' or 'client_params' will have no effect. "
+                "These parameters will be removed in a future release.",
+                category=FutureWarning,
+                stacklevel=2,
+            )
```

Comment:
Same here - `user_callback` and `client_params` don't do anything in the job classes but following the deprecation policy I don't think we can just remove them?